modules/javafx.graphics/src/main/java/com/sun/javafx/css/media/MediaRule.java

```diff
@@ -0,0 +1,130 @@
+/*
+ * Copyright (c) 2025, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.  Oracle designates this
+ * particular file as subject to the "Classpath" exception as provided
+ * by Oracle in the LICENSE file that accompanied this code.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+package com.sun.javafx.css.media;
+
+import javafx.css.StyleConverter;
+import javafx.scene.Scene;
+import java.io.DataInputStream;
+import java.io.DataOutputStream;
+import java.io.IOException;
+import java.util.List;
+
+/**
+ * CSS media rules are @-rules that contain media queries. A media query tests the "external"
+ * configuration of a {@link Scene}, and is independent of its scene graph content.
+ */
+public final class MediaRule {
+
+    private final List<MediaQuery> queries;
+    private final MediaRule parent;
+    private final int hash;
+
+    public MediaRule(List<MediaQuery> queries, MediaRule parent) {
+        this.queries = List.copyOf(queries);
+        this.parent = parent;
+        this.hash = queries.hashCode();
```

Comment:
I would rather see `this.queries.hasCode()` here

Reply:
I've removed it instead.